### PR TITLE
[M] change default content access mode list

### PR DESCRIPTION
**Pull request by jdhrh**
This satisfies a requirement of the upcoming opt-in for hosted customers to simple content access mode.

### CODE REVIEW

**Comment by jdhrh** [comment]
Looking at this spec test. I'm assuming my local config is different since it doesn't fail locally.

**Comment by nikosmoum** [comment]
@jdhrh Yes, you have to deploy candlepin in Hosted mode with the appropriate flags (`server/bin/deploy -Ha`). Certain tests only run on one mode and not on the other, or they behave differently depending on the mode.

**Comment by jdhrh** [comment]
Additionally while in here. Is it a bug that standalone can set the contentAccessModeList but not the contentAccessMode despite the exception saying it shouldn't be able to set either? The check and throw does not exist currently on the contentAccessMode configuration logic today but the exception states `"The owner content access mode and content access mode list cannot be set " +
                        "directly in standalone mode."`

**Comment by nikosmoum** [comment]
The message we return is not reflective of the actual code, but the intention is still accurate, that someone using standalone mode _shouldn't_ be changing the content access mode list (even though they can) [See](https://github.com/candlepin/candlepin/blob/f3e39f15be9dabd729005818605abb627aae64f1/server/src/main/java/org/candlepin/resource/OwnerResource.java#L920-L922), so I think we can leave it as is.

There are future plans to remove all SCA restrictions in standalone mode anyway.

**Comment by nikosmoum** [comment]
Adding in progress label since we'll need to clarify the larger SCA changes needed with a design first

**Review by nikosmoum — CHANGES_REQUESTED** [review]
While this will work for creating an Owner with the new default modeList, there's a few other places where the `ContentAccessManager.ContentAccessMode.getDefault()` method is used to check what we consider the default contentAccessModeList to be, that also need to be changed, in order for logic else where to work:
- `OwnerResource.getOwnerContentAccess()`
- 2 places in `OwnerResource.validateContentAccessModeChanges()`
- `ConsumerResource.getContentAccessForConsumer()`
- `Owner.contentAccessModeList` (default variable initialization)
- `ContentAccessManager.resolveContentAccessValue()` : This utility method seems to be used for resolving both the mode and the modeList, so my initial reaction would be to continue using it as is for wherever the mode is being resolved, but introduce a new method (e.g. `resolveContentAccessListValue`) for the places it's called from that need to resolve the modeList.

Now, applying the same change you did here to all these places is tedious and not good design. Given that throughout the system until now we've considered 'entitlement' to be the default for both mode and modeList, and now that's changed, I say we need a new method called something like `ContentAccessManager.ContentAccessMode.getListDefaultDatabaseValue()` that will return 'entitlement,org_environment' and can be used in all the places mentioned above. It returns the db value (string) because all those places I mention do call that right after they fetch the enum value. (Note: other places, that need to fetch the mode default will continue using `ContentAccessManager.ContentAccessMode.getDefault()`).

@Ceiu Do you have any better ideas here? (Keep in mind this PR need to go in ASAP, and anything that needs more thinking/design time will need to be relegated to later)

**Comment by Ceiu** [comment]
@nikosmoum @jdhrh 
While I understand the immediate need here, I also have a very strong desire to not have to deal with maintaining wonky behavior or more partial features with the SCA feature. It's been kind of a pain doing any maintenance on this feature as basically every aspect of it has gone in with a "we need it now" imperative.

That said, I think the gist of what you've outlined should work here. The change I would make is that since the default-fetching function would be returning a string, it doesn't belong on the ContentAccessMode enum, but rather on the ContentAccessManager -- something like ```ContentAccessManager.getDefaultContentAccessModeList()``` as a static function. That would serve as the drop-in replacement for all of the places called out in your comment.

As a side note, I would argue that the initializer values in owner should move to assignments in the constructor, as mixing those is kind of bad practice, and I personally find the former to be somewhat annoying.

**Comment by nikosmoum** [comment]
@jdhrh Also one more thing: this PR is for master. Once it's reviewed & merged you'll want to port the same change to candlepin-3.2-HOTIFX, since that is the y-stream you're using in Hosted right now (so the priority really is to get it to candlepin-3.2-HOTFIX).

**Comment by Ceiu** [comment]
ok to test

File: server/src/main/java/org/candlepin/resource/OwnerResource.java [hunk 1]
```diff
@@ -898,6 +898,15 @@ public OwnerDTO createOwner(@ApiParam(name = "owner", required = true) OwnerDTO
         boolean configureContentAccess = false;
 
         final String defaultContentAccess = ContentAccessMode.getDefault().toDatabaseValue();
+        final String defaultContentAccessModeList;
+        if (config.getBoolean(ConfigProperties.STANDALONE)) {
```
[the remark this answers is not in the document]
Comment (jdhrh) [reply server/src/main/java/org/candlepin/resource/OwnerResource.java]: Updated with this change.

**Review by nikosmoum — CHANGES_REQUESTED** [review]
A couple of minor changes and this should be good to merge

File: server/src/main/java/org/candlepin/controller/ContentAccessManager.java [hunk 2]
```diff
@@ -783,6 +792,18 @@ private String resolveContentAccessValue(String value, boolean resolveNull) {
         return value;
     }
 
+    private String resolveContentAccessListValue(String value, boolean resolveNull) {
```
[the remark this answers is not in the document]
Comment (jdhrh) [reply server/src/main/java/org/candlepin/controller/ContentAccessManager.java:807]: Javadocs added. Please let me know if they are sufficient.

File: server/src/main/java/org/candlepin/controller/ContentAccessManager.java [hunk 3]
```diff
@@ -127,6 +127,15 @@ public static ContentAccessMode getDefault() {
             return ENTITLEMENT;
         }
 
+        /**
+         * Fetches the default content access mode list database value
+         *
+         * @return the default content access mode list database value as a string
+         */
+        public static String getListDefaultDatabaseValue() {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/ContentAccessManager.java:135]: This should exist on ContentAccessManager, not the ContentAccessMode enum.

File: server/src/main/java/org/candlepin/controller/ContentAccessManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikosmoum) [reply server/src/main/java/org/candlepin/controller/ContentAccessManager.java:808]: ack

**Comment by nikosmoum** [comment]
@jdhrh Let's port this to candlepin-3.1-HOTIFX

**Review by Ceiu — COMMENTED** [review]
ack

**Pull request merged**